docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+Basic Usage
+===========
+
+In our opinion the best way to understand how some stuff works is by showing some examples so this section gather all knowledge form previous chapters.
+Assume that we are implementing managing countries in our system.
+Let the fun begin.
+
+Describing feature
+------------------
+First of all let's start with writing our feature file the most crucial thing here is ubiquitous language. We don't want to specify how something is done,it doesn't really matter in case of business.
+``features/addressing/managing_countries/adding_country.feature``
+.. code-block:: feature
+    @managing_countries
+    Feature: Adding a new country
+        In order to sell my goods to different countries
+        As an Administrator
+        I want to add a new country to the store
+
+        Background:
+            Given I am logged in as an administrator
+
+        @ui
+        Scenario: Adding country
+            Given I want to add a new country
+            When I choose "France"
+            And I add it
+            Then I should be notified that it has been successfully created
+            And the country "France" should appear in the store
+
+Pay special attention to form of those sentences, they are specially constructed that way to hide details like ``I click on select box and i click on France country, I see France country in box with css .table``.
+The side effect of this is that we got more generic steps and when we want to change way of testing this feature for example in domain context it will be extremely easy all we need to do is adding ``@domain`` tag and of course implement proper steps in domain context.
+
+Choosing correct suite
+----------------------
+Next thing is creating suite, in our example at the beginning we decide that it will be user interface feature.
+So this suite will be placed in ``etc/behat/suites/ui/addressing/managing_countries.yml`` and it will looks like
```

Comment:
After we are done with a feature file, we have to create a new suite for it. At the beginning we have decided that it will be a frontend/user interface feature, that is why we are placing it in "etc/behat/suites/ui/addressing/managing_countries.yml".